datafusion/physical-expr/src/expressions/dynamic_filters.rs

```diff
@@ -159,35 +139,13 @@ impl DynamicFilterPhysicalExpr {
                 )
             })?
             .clone();
-        let inner =
-            Self::remap_children(&self.children, self.remapped_children.as_ref(), inner)?;
```

Comment:
This means that `evaluate` no longer uses a version with remapped children right?

Reply:
now you evaluate based on the snapshot. snapshot is the remapped filter with your filter schema

Reply:
I feel like we're going in circles... this is the thing that was expected to be used to produce the snapshots... now we're using snapshots to produce it?

Reply:
I think the difference is that

Your version
1. we have source filter A
2. create dynamic filter a and b by different filter schema
3. you create snapshot a, b from them.
4. evaluate batches by snapshot
5. update source filter A to B
6. dynamic filter a, b remap based on source filter B when you call evaluate

My version
1. we have source filter A
2. create snapshot based on source filter A + filter schema A and B
3. evaluate batches by snapshot
4. update source filter A to B
5. create snapshot based on source filter B + filter schema A and B
6. evaluate batches by snapshot

Reply:
snapshot is actually an evaluated dynamic filter based on the schema.

Basically, what you have is just the filter expression. You provide the schema to remap the column indexes. You get yet another filter expression.